codegen/types/argument.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+
+@dataclass
+class Argument:
+    name: str
+    arg_type: str
+    map_with: Optional[str] = None
+    map_with_method: Optional[str] = None
+    include_if_null: bool = False
+
+    def generate_in_params(self) -> str:
+        return f"{self.name}: {self.arg_type}"
+
+    def generate_in_dict_inline(self) -> str:
+        value = self.name
+        if self.map_with:
+            value = f"{self.map_with}({self.name})"
+        elif self.map_with_method:
+            value = f"{self.name}.{self.map_with_method}"
+        return f'"{self.name}": {value},'
+
+    def generate_in_dict_assignment(self, dict_name: str) -> str:
+        value = self.name
+        if self.map_with:
+            value = f"{self.map_with}({self.name})"
+        elif self.map_with_method:
+            value = f"{self.name}.{self.map_with_method}"
+        return f'{dict_name}["{self.name}"] = {value}'
+
+    @property
+    def nullable(self) -> bool:
+        # This could definitely be better
+        return "Optional" in self.arg_type
```

Comment:
Any ideas? I can see that `r"^Optional\[.*\]$"` would be (pointlessly?) more robust.

You need this to generate the `if field: request['field'] = field` blocks, right? What if we used something smarter than `dict` for `request`? Something that allowed `request.set_if(field)`?

Reply:
I'll think about it. I did realize a simple improvement, though is just `self.arg_type.startswith("Optional[")

Reply:
Oh yeah, that's really all my regex was doing anyway.

I feel like I should mention that I don't think "fixing" this is important. I was mostly curious about the thoughts behind the comment 🙂